### PR TITLE
QR+PIN in Common Core

### PR DESCRIPTION
**Pull request by mipetriu**
## Proposed changes

This PR adds code for the QR+PIN feature. It creates a new parameter in the interactive token request params for the preferred auth method, which can now be QR+PIN.

Also, a new piece of deviceInfo can be returned from getDeviceInfo requests to the SSO extension - preferredAuthConfiguration.

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.m [hunk 1]
```diff
@@ -108,6 +108,10 @@ -(void)dealloc
     {
         [self.webView setNavigationDelegate:nil];
     }
+    if ([self.webView.UIDelegate isEqual:self])
```
Comment (oldalton) [inline IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.m:111]: Is UIDelegate a property from Apple or from us?

Comment (mipetriu) [reply IdentityCore/src/webview/embeddedWebview/MSIDOAuth2EmbeddedWebviewController.m:111]: Its from Apple: https://developer.apple.com/documentation/webkit/wkuidelegate?language=objc

It allows us to have control over the browser-level camera prompting behavior later in this file

File: IdentityCore/src/controllers/broker/MSIDSSOExtensionInteractiveTokenRequestController.m [hunk 2]
```diff
@@ -54,6 +54,8 @@ - (void)acquireToken:(MSIDRequestCompletionBlock)completionBlock
 {
     MSID_LOG_WITH_CTX(MSIDLogLevelInfo, self.requestParameters, @"Beginning interactive broker extension flow.");
     
+    [self addQRPINParamIfNeeded];
```
Comment (oldalton) [inline IdentityCore/src/controllers/broker/MSIDSSOExtensionInteractiveTokenRequestController.m:57]: Do we have any other places where we modify those parameters today? Maybe there's one that is more appropriate?

**Review by oldalton — APPROVED** [review]

File: IdentityCore/src/controllers/broker/MSIDSSOExtensionInteractiveTokenRequestController.m (hunk 2 above)
Comment (oldalton) [inline IdentityCore/src/controllers/broker/MSIDSSOExtensionInteractiveTokenRequestController.m:57]: Actually, I didn't see this being updated for other scenarios, like legacy broker flow and non-broker flow. Should it be updated there too?

**Pull request merged**